src/bin/handle-git-dir-change.ts

```diff
@@ -0,0 +1,221 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {execSync} from 'child_process';
+import {Changes, FileMode, FileData} from '../types';
+import {logger} from '../logger';
+import {readFile} from 'fs';
+import * as path from 'path';
+
+// data about a file in a git directory
+interface GitFileData {
+  path: string;
+  fileData: FileData;
+}
+
+// See https://git-scm.com/docs/git-diff#Documentation/git-diff.txt-git-diff-filesltpatterngt82308203
+type GitDiffStatus = 'A' | 'D' | 'M' | 'T' | 'U' | 'X' | string;
+
+class InstallationError extends Error {
+  constructor(message: string) {
+    super(message);
+    this.name = 'InstallationError';
+  }
+}
+
+/**
+ * Get the absolute path of a relative path
+ * @param {string} dir the wildcard directory containing git change, not necessarily the root git directory
+ * @returns {string} the absolute path relative to the path that the user executed the bash command in
+ */
+export function resolvePath(dir: string) {
+  const absoluteDir = path.resolve(process.cwd(), dir);
```

Comment:
If this is only used for the CLI then implicitly depending on the cwd may be ok. If this will be included later in the core (which I could potentially see), we instead want the relative root to be explicit (and provided by the caller - e.g. the CLI).

Reply:
this is explicitly for the CLI. I've intended the bin directory to be strictly for the CLI. If we want to do work for the framework-core it will not be handled here. It is a relatively small function any way.